lib/windows/app/containers/DeviceSelectorContainer.jsx

```diff
@@ -0,0 +1,198 @@
+                    errMsg += 'Please check your udev rules concerning permissions for USB devices, see https://github.com/NordicSemiconductor/nrf-udev';
+                } else if (process.platform === 'win32') {
+                    errMsg += 'Please check that a libusb-compatible kernel driver is bound to this device.';
+                }
+
+                logger.error(err.error.toString());
+                logger.error(errMsg);
+            } else {
+                logger.error(`Error while probing devices: ${err.error}`);
+            }
+        });
+
+        this.deviceLister.start();
+    }
+
+    /*
+     * Called from the templated selector.
+     * Shall receive a device definition.
+     * Resets the internal state, and calls the downstream onSelect().
+     * Note that this.onSelect() is different from this.props.onSelect(),
+     * the later one is the one that actually triggers the action.
+     */
+    onSelect(device) {
+        if (this.state.selectedSerialNumber) {
+            this.onDeselect();
+        }
+        this.setState(prev => ({ ...prev, selectedSerialNumber: device.serialNumber }));
+        this.props.onSelect(device);
+    }
+
+    /*
+     * Called from the templated selector.
+     * Resets the internal state, and calls the downstream onDeselect().
+     * Note that this.onDeselect() is different from this.props.onDeselect(),
+     * the later one is the one that actually triggers the action.
+     */
+    onDeselect() {
+        this.setState(prev => ({ ...prev, selectedSerialNumber: undefined }));
+        this.props.onDeselect();
+    }
+
+    /*
+     * Returns the JSX corresponding to a drop-down menu.
+     * Prepares some properties for the template, uses it.
+     */
+    render() {
+        let togglerText;
+        if (this.state.selectedSerialNumber) {
+            togglerText = this.state.selectedSerialNumber.toString();
+        } else if (this.state.devices.size === 0) {
+            togglerText = 'No devices available';
+        } else {
+            togglerText = 'Select device';
+        }
+
+        const displayCloseItem = Boolean(this.state.selectedSerialNumber);
+        const devices = Array.from(this.state.devices.values());
+
+        const templateProps = {
+            ...this.props,
+            togglerText,
+            displayCloseItem,
+            devices,
+            onSelect: device => { this.onSelect(device); },
+            onDeselect: () => { this.onDeselect(); },
+        };
+
+        return (
+            <DeviceSelector {...templateProps} />
+        );
+    }
+}
+
+DeviceSelectorContainer.propTypes = {
+    onSelect: PropTypes.func.isRequired,
+    onDeselect: PropTypes.func.isRequired,
+    traits: PropTypes.shape({}),
+};
+
+DeviceSelectorContainer.defaultProps = {
+    traits: undefined,
+};
+
+function mapDispatchToProps(dispatch) {
+    return {
+        onSelect: device =>
+            dispatch({
+                type: 'DEVICE_SELECTED',
```

Comment:
I would like to see these action implemented separately so app developers can follow the documentation and find these action as previously referred: https://github.com/NordicSemiconductor/pc-nrfconnect-core/wiki/API-reference#intercepting-actions-with-middleware